attribute/kv.go

```diff
@@ -82,8 +109,23 @@ func Any(k string, value interface{}) KeyValue {
 	rv := reflect.ValueOf(value)
 
 	switch rv.Kind() {
-	case reflect.Array, reflect.Slice:
+	case reflect.Array:
 		return Array(k, value)
```

Comment:
This is calling a deprecated method now. Probably this can be handled with a `Any(k, rv.Slice(0, rv.Len()).Interface()`, make a TODO?

Reply:
Yes :+1:.  I was planning to do something like this when we remove the deprecated function, but I think you make a good point we can do it now. I'll update in a bit.

Reply:
Also, this may be an non-issue if we decide to remove the `Any` function: https://github.com/open-telemetry/opentelemetry-go/issues/2179